material-ui/material-ui.d.ts

```diff
@@ -7228,7 +7228,7 @@ declare module 'material-ui/utils/withWidth' {
       mediumWidth?: number;
       resizeInterval?: number;
   }
-  export default function withWidth(options?: Options): __React.ComponentClass<any>
+  export default function withWidth<C extends __React.ComponentClass<any> | __React.StatelessComponent<any>>(options?: Options): (component: C) => C;
```

Comment:
This seems to not allow it to be used as class decorator. But this does:

    var withWidth: (options?: Options) => ClassDecorator;
    export default withWidth;

However the `"experimentalDecorators": true` tsconfig option is required for that.